src/bindings/python/bluechi/api.py

```diff
@@ -245,33 +218,38 @@ def on_agent_job_metrics(
         self.get_proxy().AgentJobMetrics.connect(callback)
 
 
-class Agent(ApiBase):
-    def __init__(self, bus: MessageBus = None, use_systembus=True) -> None:
+class Job(ApiBase):
+    def __init__(
+        self, job_path: ObjPath, bus: MessageBus = None, use_systembus=True
+    ) -> None:
         super().__init__(bus, use_systembus)
 
-        self.agent_proxy = None
+        self.job_path = job_path
+        self.job_proxy = None
 
     def get_proxy(self) -> InterfaceProxy | ObjectProxy:
-        if self.agent_proxy is None:
-            self.agent_proxy = self.bus.get_proxy(
-                BC_AGENT_DBUS_INTERFACE, BC_OBJECT_PATH
-            )
+        if self.job_proxy is None:
+            self.job_proxy = self.bus.get_proxy(BC_DBUS_INTERFACE, self.job_path)
 
-        return self.agent_proxy
+        return self.job_proxy
 
-    def create_proxy(self, local_service_name: str, node: str, unit: str) -> None:
-        self.get_proxy().CreateProxy(
-            local_service_name,
-            node,
-            unit,
-        )
+    def cancel(self) -> None:
+        self.get_proxy().Cancel()
 
-    def remove_proxy(self, local_service_name: str, node: str, unit: str) -> None:
-        self.get_proxy().RemoveProxy(
-            local_service_name,
-            node,
-            unit,
-        )
+    def get_id(self) -> UInt32:
```

Comment:
Nice to have: remove the `get_` prefix, decorate these as `@property` (https://www.programiz.com/python-programming/property) and adjust the calls

Reply:
Added the annotation and changes in a separate commit.